### PR TITLE
add cors domains parameter

### PR DESCRIPTION
**Pull request by timogoebel**
This is required to support: https://github.com/theforeman/foreman/pull/6467

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -311,6 +313,7 @@
   Stdlib::Port $dynflow_pool_size = $::foreman::params::dynflow_pool_size,
   String $jobs_service = $::foreman::params::jobs_service,
   Boolean $hsts_enabled = $::foreman::params::hsts_enabled,
+  Array[Stdlib::HTTPUrl] $cors_domains = $::foreman::params::cors_domains,
```
Comment (ekohl) [inline manifests/init.pp]: HTTPUrl -> Httpurl

File: manifests/init.pp [hunk 2]
```diff
@@ -211,6 +211,8 @@
 #
 # $hsts_enabled::                 Should HSTS enforcement in https requests be enabled
 #
+# $cors_domains::                 List of domains that show be allowed for Cross-Origin Resource Sharing
```
Comment (ekohl) [inline manifests/init.pp]: Perhaps this needs a note it's only supported with 1.22+?

File: templates/settings.yaml.erb [hunk 3]
```diff
@@ -46,6 +46,14 @@
     :type: <%= scope.lookupvar("foreman::logging_type") %>
     :layout: <%= scope.lookupvar("foreman::logging_layout") %>
 
+<% if Array(scope.lookupvar('foreman::cors_domains')).size >= 1 -%>
```
Comment (ekohl) [inline templates/settings.yaml.erb]: It's already guaranteed to be an array. Should it be `<% unless scope.lookupvar('foreman::cors_domains').empty? -%>`

Comment (timogoebel) [reply templates/settings.yaml.erb]: Ah, true. Still living in a Puppet 3 world. I'm getting old. 👴

**Comment by timogoebel** [comment]
@ekohl: Updated.

File: manifests/init.pp (hunk 1 above)
Comment (timogoebel) [reply manifests/init.pp]: Tests fail now with `unknown data type Stdlib::Httpurl`.

Comment (ekohl) [reply manifests/init.pp]: I went with https://github.com/puppetlabs/puppetlabs-stdlib#stdlibhttpurl but that's not what's actually in the code :( https://github.com/puppetlabs/puppetlabs-stdlib/pull/1001 should fix that.

**Comment by timogoebel** [comment]
Reverted the `HTTPUrl` change.

**Pull request merged**